ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -934,6 +943,15 @@ private static void createReplImportTasks(
                 tblDesc.getDatabaseName(), tblDesc.getTableName());
         return;
       }
+
+      // If the table exists and we found a valid create table event, then need to drop the table first
+      // and then create it. This case is possible if the event sequence is drop_table(t1) -> create_table(t1).
+      // We need to drop here to handle the case where the previous incremental load created the table but
+      // didn't set the last repl ID due to some failure.
+      if (x.getEventType() == DumpType.EVENT_CREATE_TABLE) {
```

Comment:
should this be done only when table exists ?
This scenario "We need to drop here to handle the case where the previous incremental load created  .. " could happen within current set of events for current incremental replication as well. Right ?

Reply:
Never mind , i see you are adding the drop for the case were table is null as well.

Reply:
No, this case is applicable only if table exists. I'm setting table=null if I add a dropTblTask here.
If table doesn't exist at the time of DAG creation and if this is create_table event, then it is guaranteed that no other previous events in this batch would create the table. So, no need of drop task. If 2 create_table events exist without drop in between, then it is invalid usage.

Reply:
This scenario "We need to drop here to handle the case where the previous incremental load created .. "
This could happen even with current batch as well. But, still we need to replace the table in this case as well as last repl ID is lower than current one.

Reply:
if its because of last repl id not set properly we should ignore the event ..and why need to create a drop task ..if the assumption is that the drop table event is in same batch ?

Reply:
That's because, we cannot decide if the existing table is result of previous replay of same event or some old table. Last repl Id is not in sequence for a table as we capture events in warehouse level. Also, there is one another case of first incremental after bootstrap, where the last repl ID may not be accurate as bootstrap dump doesn't lock the DB. We may see create table event on same table but without drop. Just to keep proper sync, it is better to re-create the table.